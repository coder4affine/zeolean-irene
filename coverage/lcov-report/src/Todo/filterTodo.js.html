<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/Todo/filterTodo.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/Todo</a> filterTodo.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/6</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { memo } from "react";
import PropTypes from "prop-types";
&nbsp;
const filterTodo = ({ filterTodos }) =&gt; {
  return (
    &lt;div
      style={{
        display: "flex",
        flexDirection: "row",
        height: 30,
        width: "100%"
      }}
    &gt;
      &lt;button
        style={{ display: "flex", flex: 1, justifyContent: "center" }}
        onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >filterTodos("all")}</span>
      &gt;
        All
      &lt;/button&gt;
      &lt;button
        style={{ display: "flex", flex: 1, justifyContent: "center" }}
        onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >filterTodos("pending")}</span>
      &gt;
        Pending
      &lt;/button&gt;
      &lt;button
        style={{ display: "flex", flex: 1, justifyContent: "center" }}
        onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >filterTodos("completed")}</span>
      &gt;
        Completed
      &lt;/button&gt;
    &lt;/div&gt;
  );
};
&nbsp;
filterTodo.propTypes = {};
&nbsp;
export default memo(filterTodo);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Mar 03 2019 20:56:55 GMT+0530 (India Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
