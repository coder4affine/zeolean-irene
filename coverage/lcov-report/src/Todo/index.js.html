<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/Todo/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/Todo</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.89% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.11% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/18</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { PureComponent } from "react";
// import PropTypes from "prop-types";
import TodoForm from "./todoForm";
import TodoList from "./todoList";
import FilterTodo from "./filterTodo";
// Import FilterTodo here
&nbsp;
// https://demos.scotch.io/visual-guide-to-css3-flexbox-flexbox-playground/demos/
&nbsp;
export default class index extends PureComponent {
  static propTypes = {};
&nbsp;
  state = {
    error: false,
    todo: "",
    todos: [],
    status: "all"
  };
&nbsp;
  onChange = e =&gt; {
    this.setState({ todo: e.target.value });
  };
&nbsp;
  onAddTodo = e =&gt; {
    e.preventDefault();
    const { todo, todos } = this.state;
    if (!todo) {
      this.setState({ error: "Required" });
      return;
    }
    this.setState({
      todos: [
        ...todos,
        { id: new Date().valueOf(), text: todo, isDone: false }
      ],
      todo: ""
    });
  };
&nbsp;
  onCompleteTask = <span class="fstat-no" title="function not covered" >id</span> =&gt; {
    const { todos } = <span class="cstat-no" title="statement not covered" >this.state;</span>
    const index = <span class="cstat-no" title="statement not covered" >todos.findIndex(<span class="fstat-no" title="function not covered" >x </span>=&gt; <span class="cstat-no" title="statement not covered" >x.id === id)</span>;</span>
    const newTodos = <span class="cstat-no" title="statement not covered" >[</span>
      ...todos.slice(0, index),
      { ...todos[index], isDone: !todos[index].isDone },
      ...todos.slice(index + 1)
    ];
<span class="cstat-no" title="statement not covered" >    this.setState({ todos: newTodos });</span>
  };
&nbsp;
  deleteTodo = <span class="fstat-no" title="function not covered" >id</span> =&gt; {
    const { todos } = <span class="cstat-no" title="statement not covered" >this.state;</span>
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      todos: todos.filter(<span class="fstat-no" title="function not covered" >x </span>=&gt; <span class="cstat-no" title="statement not covered" >x.id !== id)</span>
    });
  };
&nbsp;
  filterTodos = status =&gt; {
    // const { todos } = this.state;
    // this.setState({
    //   todos: todos.filter(x =&gt; {
    //     if (status === "pending") {
    //       return !x.isDone;
    //     } else if (status === "completed") {
    //       return x.isDone;
    //     } else {
    //       return true;
    //     }
    //   })
    // });
    this.setState({ status });
  };
&nbsp;
  render() {
    console.log("render");
    const { error, todo, todos, status } = this.state;
&nbsp;
    return (
      &lt;div
        style={{
          display: "flex",
          flexDirection: "column",
          alignItems: "center",
          maxWidth: 1024,
          height: "100vh"
        }}
      &gt;
        &lt;div
          style={{
            display: "flex",
            flex: 1,
            flexDirection: "column",
            width: "100%",
            alignItems: "center"
          }}
        &gt;
          &lt;h1&gt;My To-do&lt;/h1&gt;
          &lt;TodoForm
            onAddTodo={this.onAddTodo}
            todo={todo}
            error={error}
            onChange={this.onChange}
          /&gt;
          &lt;TodoList
            todos={todos}
            status={status}
            onCompleteTask={this.onCompleteTask}
            deleteTodo={this.deleteTodo}
          /&gt;
        &lt;/div&gt;
        &lt;FilterTodo filterTodos={this.filterTodos} /&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Mar 03 2019 20:56:55 GMT+0530 (India Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
