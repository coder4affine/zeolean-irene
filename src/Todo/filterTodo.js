import React, { memo } from "react";
import PropTypes from "prop-types";
import Tooltip from "../component/Tooltip";

const filterTodo = ({ filterTodos }) => {
  console.log("filterTodo");
  return (
    <div
      style={{
        display: "flex",
        flexDirection: "row",
        height: 30,
        width: "100%"
      }}
    >
      <button
        style={{ display: "flex", flex: 1, justifyContent: "center" }}
        onClick={() => filterTodos("all")}
      >
        <Tooltip text="Display all todos">All</Tooltip>
      </button>
      <button
        style={{ display: "flex", flex: 1, justifyContent: "center" }}
        onClick={() => filterTodos("pending")}
      >
        <Tooltip text="Display pending todos">Pending</Tooltip>
      </button>
      <button
        style={{ display: "flex", flex: 1, justifyContent: "center" }}
        onClick={() => filterTodos("completed")}
      >
        <Tooltip text="Display completed todos">Completed</Tooltip>
      </button>
    </div>
  );
};

filterTodo.propTypes = {
  filterTodos: PropTypes.func.isRequired
};

export default memo(filterTodo);
