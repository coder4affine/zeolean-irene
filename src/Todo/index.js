import React, { PureComponent } from "react";
// import PropTypes from "prop-types";
import TodoForm from "./todoForm";
import TodoList from "./todoList";
// Import FilterTodo here

// https://demos.scotch.io/visual-guide-to-css3-flexbox-flexbox-playground/demos/

export default class index extends PureComponent {
  static propTypes = {};

  state = {
    error: false,
    todo: "",
    todos: [],
    status: "all"
  };

  onChange = e => {
    this.setState({ todo: e.target.value });
  };

  onAddTodo = e => {
    e.preventDefault();
    const { todo, todos } = this.state;
    if (!todo) {
      this.setState({ error: "Required" });
      return;
    }
    this.setState({
      todos: [
        ...todos,
        { id: new Date().valueOf(), text: todo, isDone: false }
      ],
      todo: ""
    });
  };

  onCompleteTask = id => {
    const { todos } = this.state;
    const index = todos.findIndex(x => x.id === id);
    const newTodos = [
      ...todos.slice(0, index),
      { ...todos[index], isDone: !todos[index].isDone },
      ...todos.slice(index + 1)
    ];
    this.setState({ todos: newTodos });
  };

  deleteTodo = id => {
    const { todos } = this.state;
    this.setState({
      todos: todos.filter(x => x.id !== id)
    });
  };

  filterTodos = status => {
    // filter
  };

  render() {
    console.log("render");
    const { error, todo, todos } = this.state;

    return (
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          alignItems: "center",
          maxWidth: 1024
        }}
      >
        <h1>My To-do</h1>
        <TodoForm
          onAddTodo={this.onAddTodo}
          todo={todo}
          error={error}
          onChange={this.onChange}
        />
        <TodoList
          todos={todos}
          onCompleteTask={this.onCompleteTask}
          deleteTodo={this.deleteTodo}
        />

        {/* Create new Component FilterTodo */}
      </div>
    );
  }
}
