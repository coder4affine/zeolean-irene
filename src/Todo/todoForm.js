import React, { memo } from "react";
import PropTypes from "prop-types";
import test from "../HOC/test";

const todoForm = ({ onAddTodo, todo, error, onChange }) => {
  console.log("todoForm");
  return (
    <form onSubmit={onAddTodo}>
      <div style={{ display: "flex" }}>
        <div style={{ display: "flex", flexDirection: "column" }}>
          <input
            style={{ height: 24 }}
            type="text"
            value={todo}
            onChange={onChange}
            placeholder="Write your to-do"
          />
          {error && <span style={{ fontSize: 11, color: "red" }}>{error}</span>}
        </div>
        <input style={{ height: 30 }} type="submit" value="Add To-DO" />
      </div>
    </form>
  );
};

todoForm.defaultProps = {
  todo: ""
};

todoForm.propTypes = {
  onAddTodo: PropTypes.func.isRequired,
  todo: PropTypes.string,
  error: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired
};

export default memo(test(todoForm));
