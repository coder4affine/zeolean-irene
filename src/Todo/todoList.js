import React, { memo } from 'react';
import PropTypes from 'prop-types';

const todoList = ({ todos, onCompleteTask, deleteTodo, status }) => {
  const filteredTodos = todos.filter(x => {
    if (status === 'pending') {
      return !x.isDone;
    }
    if (status === 'completed') {
      return x.isDone;
    }
    return true;
  });
  return (
    <div
      style={{
        overflow: 'auto',
        width: '100%',
      }}
    >
      {filteredTodos.map(item => (
        <div style={{ display: 'flex', margin: 10 }} key={item.id}>
          <input
            type="checkbox"
            defaultChecked={item.isDone}
            value={item.isDone}
            onClick={() => onCompleteTask(item.id)}
          />
          <span
            style={{
              display: 'flex',
              flex: 1,
              padding: '0 10px',
              textDecoration: item.isDone ? 'line-through' : 'none',
            }}
          >
            {item.text}
          </span>
          <button style={{ height: 30 }} onClick={() => deleteTodo(item.id)}>
            Delete
          </button>
        </div>
      ))}
    </div>
  );
};

todoList.propTypes = {
  todos: PropTypes.array.isRequired,
  onCompleteTask: PropTypes.func.isRequired,
  deleteTodo: PropTypes.func.isRequired,
  status: PropTypes.string.isRequired,
};

export default memo(todoList);
