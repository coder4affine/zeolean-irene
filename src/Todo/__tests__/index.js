import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Todo from '../index';
import TodoForm from '../todoForm';
import TodoList from '../todoList';
import FilterTodo from '../filterTodo';

// https://airbnb.io/enzyme/

// https://jestjs.io/docs/en/getting-started

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {};

  const wrapper = shallow(<Todo {...props} />);

  return {
    props,
    wrapper,
  };
}

describe('snapshot', () => {
  it('snapshot of todo', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('test todo component', () => {
  it('check for the components', () => {
    const { wrapper } = setup();
    expect(wrapper.find('h1').text()).toBe('My To-do');
    const state = wrapper.state();

    expect(wrapper.find(TodoForm).props()).toEqual({
      error: state.error,
      onAddTodo: expect.any(Function),
      onChange: expect.any(Function),
      todo: state.todo,
    });

    expect(wrapper.find(TodoList).props()).toEqual({
      todos: state.todos,
      status: state.status,
      onCompleteTask: expect.any(Function),
      deleteTodo: expect.any(Function),
    });

    expect(wrapper.find(FilterTodo).props()).toEqual({
      filterTodos: expect.any(Function),
    });
  });

  it('check TodoForms onChange function', () => {
    const { wrapper } = setup();
    const Form = wrapper.find(TodoForm);
    Form.props().onChange({ target: { value: 'my todos' } });
    expect(wrapper.state('todo')).toEqual('my todos');
  });

  it('check TodoForms onChange function', () => {
    const { wrapper } = setup();
    const Form = wrapper.find(TodoForm);
    Form.props().onAddTodo({ preventDefault: jest.fn() });
    expect(wrapper.state('error')).toEqual('Required');
    wrapper.setState({ todo: 'New todo' });
    Form.props().onAddTodo({ preventDefault: jest.fn() });
    expect(wrapper.state('todos').length).toEqual(1);
    wrapper.setState({ todo: 'New todo1' });
    Form.props().onAddTodo({ preventDefault: jest.fn() });
    expect(wrapper.state('todos').length).toEqual(2);
  });

  it('check FilterTodo component', () => {
    const { wrapper } = setup();
    const filterTodo = wrapper.find(FilterTodo);
    expect(wrapper.state('status')).toEqual('all');
    filterTodo.props().filterTodos('pending');
    expect(wrapper.state('status')).toEqual('pending');
  });
});
