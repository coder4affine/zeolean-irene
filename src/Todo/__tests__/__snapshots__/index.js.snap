// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot snapshot of todo 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <index />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
              "width": "100%",
            }
          }
        >
          <h1>
            My To-do
          </h1>
          <UNDEFINED
            error={false}
            onAddTodo={[Function]}
            onChange={[Function]}
            todo=""
          />
          <UNDEFINED
            deleteTodo={[Function]}
            onCompleteTask={[Function]}
            status="all"
            todos={Array []}
          />
        </div>,
        <UNDEFINED
          filterTodos={[Function]}
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "height": "100vh",
        "maxWidth": 1024,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              My To-do
            </h1>,
            <UNDEFINED
              error={false}
              onAddTodo={[Function]}
              onChange={[Function]}
              todo=""
            />,
            <UNDEFINED
              deleteTodo={[Function]}
              onCompleteTask={[Function]}
              status="all"
              todos={Array []}
            />,
          ],
          "style": Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 1,
            "flexDirection": "column",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "My To-do",
            },
            "ref": null,
            "rendered": "My To-do",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": false,
              "onAddTodo": [Function],
              "onChange": [Function],
              "todo": "",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "deleteTodo": [Function],
              "onCompleteTask": [Function],
              "status": "all",
              "todos": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            },
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "filterTodos": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flex": 1,
                "flexDirection": "column",
                "width": "100%",
              }
            }
          >
            <h1>
              My To-do
            </h1>
            <UNDEFINED
              error={false}
              onAddTodo={[Function]}
              onChange={[Function]}
              todo=""
            />
            <UNDEFINED
              deleteTodo={[Function]}
              onCompleteTask={[Function]}
              status="all"
              todos={Array []}
            />
          </div>,
          <UNDEFINED
            filterTodos={[Function]}
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "height": "100vh",
          "maxWidth": 1024,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                My To-do
              </h1>,
              <UNDEFINED
                error={false}
                onAddTodo={[Function]}
                onChange={[Function]}
                todo=""
              />,
              <UNDEFINED
                deleteTodo={[Function]}
                onCompleteTask={[Function]}
                status="all"
                todos={Array []}
              />,
            ],
            "style": Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "My To-do",
              },
              "ref": null,
              "rendered": "My To-do",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": false,
                "onAddTodo": [Function],
                "onChange": [Function],
                "todo": "",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "type": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "deleteTodo": [Function],
                "onCompleteTask": [Function],
                "status": "all",
                "todos": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "type": [Function],
              },
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "filterTodos": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "compare": null,
            "type": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
